api/include/opentelemetry/logs/logger.h

```diff
@@ -43,81 +60,105 @@ class Logger
   // the specification. virtual nostd::string_view getName() = 0;
 
   /**
-   * Each of the following overloaded log(...) methods
+   * Each of the following overloaded Log(...) methods
    * creates a log message with the specific parameters passed.
    *
+   * @param timestamp the timestamp the log record was created.
    * @param name the name of the log event.
    * @param severity the severity level of the log event.
    * @param message the string message of the log (perhaps support std::fmt or fmt-lib format).
    * @param record the log record (object type LogRecord) that is logged.
    * @param attributes the attributes, stored as a 2D list of key/value pairs, that are associated
-   * with this log.
+   * wtih the log event.
+   * @param trace_id the trace id associated with the log event.
+   * @param span_id the span id associate with the log event.
+   * @param trace_flags the trace flags associated with the log event.
    * @throws No exceptions under any circumstances.
    */
 
-  /* The below method is a logging statement that takes in a LogRecord.
-   * A default LogRecord that will be assigned if no parameters are passed to Logger's .log() method
-   * which should at minimum assign the trace_id, span_id, and timestamp
+  /**
+   * The base Log(...) method that all other Log(...) overloaded methods will eventually call,
+   * in order to create a log record.
+   *
+   * Note this takes in a KeyValueIterable for the resource and attributes fields.
    */
-  virtual void log(const LogRecord &record) noexcept = 0;
-
-  /** Overloaded methods for unstructured logging **/
-  inline void log(nostd::string_view message) noexcept
+  virtual void Log(core::SystemTimestamp timestamp,
+                   Severity severity,
+                   nostd::string_view name,
+                   nostd::string_view body,
+                   const common::KeyValueIterable &resource,
+                   const common::KeyValueIterable &attributes,
+                   trace::TraceId trace_id,
+                   trace::SpanId span_id,
+                   trace::TraceFlags trace_flags) noexcept = 0;
+
+  /*** Overloaded methods for KeyValueIterables ***/
+  /**
+   * The secondary base Log(...) method that all other Log(...) overloaded methods except the one
+   * above will eventually call,  in order to create a log record.
+   *
+   * Note this takes in template types for the resource and attributes fields.
+   */
+  template <class T,
+            class U,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<T>::value> * = nullptr,
+            nostd::enable_if_t<common::detail::is_key_value_iterable<U>::value> * = nullptr>
+  void Log(core::SystemTimestamp timestamp,
+           Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           const T &resources,
```

Comment:
Should this be `resource` ?
```suggestion
           const T &resource,
```